core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -633,6 +633,13 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
       _ <- (l.await *> ((self provide r) to p)).fork
     } yield l.succeed(()) *> p.await
 
+  /**
+   * Returns a new effect where the error channel has been merged into the
+   * success channel to their common combined type.
+   */
+  final def merge[A1 >: A](implicit ev: E <:< A1): URIO[R, A1] =
```

Comment:
You could use `CanFail` here. It doesn't make sense to `merge` a `ZIO[R, Nothing, A]`.

Reply:
Yeah, good point. Thanks.